tests/integration/plugins/test_gradle.py

```diff
@@ -0,0 +1,147 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import subprocess
+import textwrap
+from pathlib import Path
+
+import pytest
+import yaml
+from craft_parts import LifecycleManager, Step
+from craft_parts.infos import ProjectInfo
+
+
+@pytest.fixture
+def local_proxy_url():
+    conf_file = Path("proxy.conf")
+    conf_file.write_text(
+        """
+Port 8888
+Timeout 600
+DefaultErrorFile "/usr/share/tinyproxy/default.html"
+StatFile "/usr/share/tinyproxy/stats.html"
+LogFile "/var/log/tinyproxy/tinyproxy.log"
+LogLevel Info
+PidFile "/run/tinyproxy/tinyproxy.pid"
+MaxClients 100
+Allow 127.0.0.1
+Allow ::1
+ViaProxyName "tinyproxy"
+    """,
+        encoding="utf-8",
+    )
+    proc = subprocess.Popen(["sudo", "tinyproxy", "-d", str(conf_file)])
+    yield "http://localhost:8888"
+    proc.kill()
+
+
+@pytest.fixture
+def use_gradlew(request):
+    if request.param:
+        yield request.param
+        return
+    source_location = Path(__file__).parent / "test_gradle"
+    gradlew_file = source_location / "gradlew"
+    gradlew_file = gradlew_file.rename(f"{source_location}/gradlew.backup")
+    yield request.param
+    gradlew_file.rename(f"{source_location}/gradlew")
```

Comment:
instead of doing this, I think the cleaner way would be to copy the original sources to a place inside `new_dir`, and remove the file if `use_gradlew` is false.
Otherwise, if the test dies (e.g. the process segfaults) before reaching this `rename()` things will get ugly